ethereum/rpc/namespaces/debug/api.go

```diff
@@ -112,6 +112,85 @@ func (a *API) TraceTransaction(hash common.Hash, config *evmtypes.TraceConfig) (
 	return decodedResult, nil
 }
 
+func (a *API) TraceBlockByNumber(height rpctypes.BlockNumber, config *evmtypes.TraceConfig) ([]*TxTraceResult, error) {
+	a.logger.Debug("debug_traceBlockByNumber", "height", height)
+	if height == 0 {
+		return nil, errors.New("genesis is not traceable")
+	}
+	// Get Tendermint Block
+	resBlock, err := a.backend.GetTendermintBlock(height)
+	if err != nil {
+		return nil, err
+	}
+
+	txsLength := len(resBlock.Block.Txs)
+
+	if txsLength == 0 {
+		// If there are no transactions return empty array
+		return []*TxTraceResult{}, nil
+	}
+
+	var (
+		results = make([]*TxTraceResult, txsLength)
+		wg = new(sync.WaitGroup)
+		jobs = make(chan *TxTraceTask, txsLength)
+	)
+
+	threads := runtime.NumCPU()
+	if threads > txsLength {
+		threads = txsLength
+	}
+	wg.Add(threads)
+	for th := 0; th < threads; th++ {
+		go func() {
+			defer wg.Done()
+			// Fetch and execute the next transaction trace tasks
+			for task := range jobs {
+				tx, err := a.clientCtx.TxConfig.TxDecoder()(resBlock.Block.Txs[task.Index])
+				if err != nil {
+					a.logger.Debug("tx not found", "hash", resBlock.Block.Txs[task.Index].Hash())
```

Comment:
this should be `Error` log
```suggestion
					a.logger.Error("failed to decode tx", "hash", resBlock.Block.Txs[task.Index].Hash(), "error", error.Error())
```